templates/base/types.ts.hbs

```diff
@@ -28,6 +28,22 @@ export interface {{makeRequestName @key}} {
 {{/if}}
 }
 {{/each}}
+{{#each auth}}
+export interface {{capitalize @key}}UserData {
+  type: "{{@key}}";
+  id: string;
```

Comment:
in other files you seem to assume that "username" exists in UserData. You should either have these {Auth}UserData interfaces implement an interface that contains "type", "id", and "username" or at the very least ensure "username" exists outside of the if blocks below.